### ISSUE TITLE
Track indentation in yaml.Node

### ISSUE THREAD

**Issue by rbren** [go-yaml/yaml#899]
I'm really enjoying the ability to use `yaml.Node` to modify a YAML document while mostly preserving comments and formatting. The one big gap is the inability to preserve indentation.

I've found a couple related issues, but they don't quite get at the problem:

* https://github.com/go-yaml/yaml/issues/720 (specifically about list indentation)
* https://github.com/go-yaml/yaml/issues/709 (mainly about a bug in preserving comments)

I would be happy to take a stab at implementing this if y'all would accept a PR (and maybe provide some guidance).

**Comment by Justinzobel** [comment]
Yes the inability to preserve existing indentation is certainly a problem.
